linked_list/linked_list.py

```diff
@@ -1,100 +1,198 @@
 
 # Defines a node in the singly linked list
-class Node:
+from email.base64mime import header_encode
+from subprocess import list2cmdline
+
 
-    def __init__(self, value, next_node = None):
+class Node:
+    def __init__(self, value, next_node=None):
         self.value = value
         self.next = next_node
 
 # Defines the singly linked list
+
+
 class LinkedList:
     def __init__(self):
-      self.head = None # keep the head private. Not accessible outside this class
+        self.head = None  # keep the head private. Not accessible outside this class
 
     # returns the value in the first node
     # returns None if the list is empty
-    # Time Complexity: ?
-    # Space Complexity: ?
+    # Time Complexity: O(1)
+    # Space Complexity: O(1)
     def get_first(self):
-        pass
+        if self.head is None:
+            return None
 
+        return self.head.value
 
     # method to add a new node with the specific data value in the linked list
     # insert the new node at the beginning of the linked list
-    # Time Complexity: ?
-    # Space Complexity: ?
+    # Time Complexity: O(1)
+    # Space Complexity: O(1)
+
     def add_first(self, value):
-        pass
+        if self.head is None:
+            self.head = self.tail = Node(value)
+        else:
+            new_node = Node(value, self.head)
+            self.head.previous = new_node
+            self.head = new_node

```

Comment:
Can be shortened a bit

```suggestion
        self.head = Node(value, self.head)
        if self.tail is None:
            self.tail = self.head
```